documentation/CAMARA-Security-Interoperability.md

```diff
@@ -93,48 +93,52 @@ The client MUST authenticate with the authorization server as described in [Clie
+OIDC also defines that the parameter acr_values is OPTIONAL and does not specify possible values. This leads to interoperability issues.
 
-## The Scope Parameter
+This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
 
-Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects.
+This document defines that Camara Clients SHOULD not use the acr_values parameter. 
 
-### Missing "openid" scope
+> To foster interoperability a future version of this document might define values for the acr_values parameter acceptable in Camara.
 
-[OIDC Core Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) states the following about the value of scope.
-> REQUIRED. OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the behavior is entirely unspecified. Other scope values MAY be present. ...
+## Access Token Request
 
-This document defines the following error handling for a missing "openid" value in scope. 
-Please refer to [Authentication Error Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthError).
+This CAMARA document makes the scope parameter REQUIRED for the OAuth2 Client Credentials Grant.
 
-If "openid" is missing in the scope value, the Authorization Server returns an HTTP response code of 400 (Bad Request) and an error invalid_request.
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
 
-### Handling of acr_values
+## The Scope Parameter
 
-OIDC specifies in [Mandatory to Implement Features for All OpenID Providers](https://openid.net/specs/openid-connect-core-1_0.html#ServerMTI) that OpenId Providers MUST implement support for acr_values.
+Scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services. The scope values must be documented in the API OAS files by all Camara API subprojects.
 
-> Authentication Context Class Reference
->>    OPs MUST support requests for specific Authentication Context Class Reference values via the acr_values parameter, as defined in Section 3.1.2. (Note that the minimum level of support required for this parameter is simply to have its use not result in an error.)
+### Purpose as part of scope
 
-OIDC also defines that the parameter acr_values is OPTIONAL and does not specify possible values. This leads to interoperability issues.
+To improve interoperablitiy this document RECOMMENDS that API Working groups define scope values following this scheme:
 
-This documents defines that Camara OpenId Providers MUST ignore the parameter acr_values. 
+`dpv:<dpvValue>#<technical_scope>`
 
```

Comment:
The dpvValue must never be part of the scope definition in the API spec. 
Some operators might allow for advertizing, others not.
Also it would lead to the fact that when requesting multiple scopes, different dpvValues values (purpose)  could be part of one access token request. This cannot be handled. 
When - during the authorization request , a consent check happens at that point in time. and the generated access token will contain the list of permitted scopes . Now assume that these scopes e.g. indicate:
advertizement#qod-session-read and serviceprovisioning#qod-session-write and serviceprovisioning#qod-session-read
When the service gets this access token as part of the api call, it needs to know in addition, what is the purpose of this, otherwise no chance to judge.
Conclusion: It is actually NOT recommended to include purpose values in the scopes of the 'API spec for various reasons.

Reply:
I agree with your analysis.

In some of the initial proposals, the purpose was sent in a separate "purpose" query parameter in the /authorize call. The purpose was moved as an additional "scope" attribute to avoid deviating too much from the standard. It never has to be included in the OpenAPI specs because the same API can be consumed for many different purposes. And when you request a token, you need to send the purpose.

The commit a7156c0a270e152d696af2c17b3944d9178a54a5 will be problematic, because "each single scope value" means it can be more than one purpose. However, [the 0.1.0 docs](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#applying-purpose-concept-in-the-authorization-request, but not multiple options) reflects the current agreement that it has to be either _one_ scope or _one_ api.

We need to review it to find a long-term solution that allows us to request multiple scopes for a given purpose, etc. But since it took us +9 months to reach the short-term agreement in #32, I'd strongly recommend to revert this commit and reflect what is currently stated in 0.1.0.

Reply:
I amended that section. Resolving this. Please review the amended section.